throughs/map.js

```diff
@@ -1,6 +1,7 @@
 var ThroughStream = require('./through')
 
 function MapStream(fn) {
+  if (!(this instanceof MapStream)) return new MapStream(fn)
```

Comment:
This actually fixed map.

Reply:
why not use the same pattern as through?

Reply:
Ah, we could, but I'm just fixing this bug now. I'll actually change the class-related code style a lot later.